src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/RoleAssignmentFilteringResult.java

```diff
@@ -28,4 +28,29 @@ public boolean atLeastOneCaseRoleExists() {
             .map(Pair::getKey)
             .anyMatch(RoleAssignment::isCaseRoleAssignment);
     }
+
+
+    public boolean hasGrantTypeExcluded() {
+        return getRoleAssignments().stream()
+            .anyMatch(roleAssignment -> roleAssignment.getGrantType().equals(GrantType.EXCLUDED.name()));
+    }
+
+    public RoleAssignmentFilteringResult getBasicAndStandardGrantTypeRoles() {
+        return new RoleAssignmentFilteringResult(roleAssignmentRoleMatchingResults
+            .stream()
+            .filter(pair ->
+                pair.getKey().getGrantType().equals(GrantType.BASIC.name())
+            || pair.getKey().getGrantType().equals(GrantType.STANDARD.name()))
+            .collect(Collectors.toList()));
+    }
+
+    @Override
+    public boolean hasNext() {
+        return false;
+    }
+
+    @Override
+    public Object next() {
+        return null;
+    }
```

Comment:
no point in implementing Iterator if you are not using it. Either we use it in: 
```
        for (RoleAssignment roleAssignment : filteringResults.getRoleAssignments()) {
```
by changing to 
```for (RoleAssignment roleAssignment : filteringResults) {
or we remove the Iterator interface

Reply:
removed iterator implementation